feature/match-navigation

Mostafa-khaled21 opened the pull request.

Mostafa-khaled21 committed:
add match pager route and update navigation

Mostafa-khaled21 committed:
Update loading state in match pager

This commit introduces a new `isLoading` state to the `MatchPagerScreen` and `MatchPagerViewModel`.

The `isLoading` state is set to `true` when the user clicks the "Next" button on the last page of the pager, and then set back to `false` after a short delay. This provides visual feedback to the user while the matching process is being finalized.

mohamedebrahem13 commented on presentation/match/src/main/java/com/giraffe/match/screen/match_pager/MatchPagerRoute.kt:
```diff
@@ -0,0 +1,20 @@
+    onFinish: () -> Unit
+) {
+    composable<MatchRoutePager> {
+        MatchPagerScreen(
```
There is a missing screen for the strat destination for match called match screen this should navigate to the match pager screen

Mohamed-Ashraf-2210 commented on presentation/match/src/main/java/com/giraffe/match/screen/match_pager/MatchPagerViewModel.kt:
```diff
@@ -35,6 +36,9 @@ class MatchPagerViewModel @Inject constructor() : ViewModel() {
                 _state.value = _state.value.copy(currentPage = _state.value.currentPage + 1)
             } else {
+                _state.value = _state.value.copy(isLoading = true)
+                delay(500)
```
why put this delay and change this hard code to make this function more readable

Mostafa-khaled21 replied:
The loading button in the design, so I added a delay so that the loading would appear in the application

HendSayed25 commented on presentation/match/src/main/java/com/giraffe/match/navigation/MatchNavGraph.kt:
```diff
@@ -19,8 +21,14 @@ internal fun MatchNavGraph(
     ) {
+
+        matchPagerRoute(
+            onBackClick = { navController.popBackStack() },
```
This way is better 
```suggestion
            onBackClick =  navController::popBackStack,
```

HendSayed25 commented on presentation/match/src/main/java/com/giraffe/match/navigation/MatchNavGraph.kt:
```diff
@@ -19,8 +21,14 @@ internal fun MatchNavGraph(
+
+        matchPagerRoute(
+            onBackClick = { navController.popBackStack() },
+            onFinish = { navController.navigateToMatchResult() }
```
also here 
```suggestion
            onFinish =  navController::navigateToMatchResult
```

Mostafa-khaled21 committed:
Refactor match navigation and add loading delay constant

This commit introduces the following changes:
- Adds a `LOADING_DELAY` constant in `MatchPagerViewModel`.
- Modifies `MatchPagerRoute.kt` to include `navigateToMatchPager` and updates `matchPagerRoute`.
- Updates `MatchNavGraph.kt` to use new navigation functions and set `MatchRouteStart` as the start destination.
- Renames `onBackButtonClick` to `onBackClick` in `MatchResultRoute.kt`.
- Adds a new file `MatchScreenRoute.kt` with `MatchRouteStart` and `matchRouteStart`.

Mostafa-khaled21 committed:
remove loading state in match pager screen

Mostafa-khaled21 committed:
feat: manage bottom bar visibility in match module

This commit introduces logic to control the visibility of the bottom navigation bar within the match module.

Specifically, the following changes were made:
- Added logic to `MatchNavGraph.kt` to show the bottom bar only for the `MatchRouteStart` destination.
- Updated `navigateToMatchResult` in `MatchResultRoute.kt` to pop up to `MatchRoutePager` when navigating to the results screen, effectively removing the pager from the back stack.
- Passed the `onShowBottomBarChange` callback to `MatchNavGraph` in `MatchApiImp.kt` to allow it to communicate bottom bar visibility changes.
- Renamed `matchRoute` to `matchRouteResult` in `MatchResultRoute.kt` and its usages for clarity.
